uui-components/src/overlays/Dropdown.tsx

```diff
@@ -127,11 +131,16 @@ export class Dropdown extends React.Component<DropdownProps, DropdownState> {
     }
 
     private handleMouseEnter = (e: Event) => {
-        this.handleOpenedChange(true);
+        this.setOpenDropdownTimer();
     }
 
     private handleMouseLeave = (e: MouseEvent) => {
-        this.handleOpenedChange(false);
+        if (this.props.openDelay && !this.props.closeDelay) {
+            this.handleOpenedChange(false);
+            this.clearOpenDropdownTimer();
```

Comment:
It seems "clearOpenDropdownTimer" should be always executed without conditions, because "open" timer is always set on mouse hover (even if the delay = 0, it's still asynchronous and not executed immediately). I.e. I would rewrite it like this:

```
private handleMouseLeave = (e: MouseEvent) => {
    this.clearOpenDropdownTimer();
    if (!this.props.closeDelay) {
        this.handleOpenedChange(false);
    } else {
        this.setCloseDropdownTimer();
    }
}
```